[2024-11-07 14:24:19,254][root][INFO] - ---------------------------------------------------------------
[2024-11-07 14:24:19,255][root][INFO] - Env:
CUDA_VISIBLE_DEVICES: null
PYTHONOPTIMIZE: null

[2024-11-07 14:24:19,256][root][INFO] - Command line args:
ansatz.alpha=6
[2024-11-07 14:24:19,257][root][INFO] - ---------------------------------------------------------------
[2024-11-07 14:24:19,259][root][INFO] - Run config:
device: '4'
solver_fn:
  _target_: netket.optimizer.solver.cholesky
lr: 0.001
diag_shift: 1.0e-10
n_iter: 2000
chunk_size_jac: 2
chunk_size_vmap: 1
save_every: 10
base_path: /scratch/.amisery/grad_sample/
model:
  _target_: grad_sample.models.ising.TFI
  L: 3
  h: 1.0
ansatz:
  _target_: netket.models.RBM
  alpha: 6
  param_dtype: complex
task:
  _target_: grad_sample.tasks.fullsum_analysis.FullSumPruning
  deltadep: false

[2024-11-07 14:24:19,260][root][INFO] - ---------------------------------------------------------------
[2024-11-07 14:24:24,264][jax._src.xla_bridge][INFO] - Unable to initialize backend 'rocm': module 'jaxlib.xla_extension' has no attribute 'GpuAllocatorConfig'
[2024-11-07 14:24:24,267][jax._src.xla_bridge][INFO] - Unable to initialize backend 'tpu': INTERNAL: Failed to open libtpu.so: libtpu.so: cannot open shared object file: No such file or directory
[2024-11-07 14:24:37,701][root][CRITICAL] - RESOURCE_EXHAUSTED: Out of memory while trying to allocate 2319450112 bytes.
Traceback (most recent call last):
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/main.py", line 50, in main
    task()
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/tasks/fullsum_analysis.py", line 77, in __call__
    in_idx, fid_vals, dp_dist_ev, infid_ev = self.prune()
                                             ^^^^^^^^^^^^
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/tasks/fullsum_analysis.py", line 236, in prune
    in_idx, fid_vals = self.get_change(jnp.arange(self.jacobian_orig.shape[0])) #compute infidelity change on entire basis
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/tasks/fullsum_analysis.py", line 162, in get_change
    fid_vals = self.vmap_change(indices)
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/scratch/.amisery/netket/lib/python3.11/site-packages/netket/jax/_vmap_chunked.py", line 19, in _eval_fun_in_chunks
    y = vmapped_fun(*args, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/tasks/fullsum_analysis.py", line 185, in compute_change
    dp_sol = self.get_new_dp(jac_new, Hloc_new)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/tasks/fullsum_analysis.py", line 222, in get_new_dp
    new_qgt = (jac_new.transpose().conj() @ jac_new) + self.diag_shift*jnp.eye(jac_new.shape[1])
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
  File "/scratch/.amisery/netket/lib/python3.11/site-packages/jax/_src/numpy/array_methods.py", line 1050, in op
    return getattr(self.aval, f"_{name}")(self, *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/scratch/.amisery/netket/lib/python3.11/site-packages/jax/_src/numpy/array_methods.py", line 573, in deferring_binary_op
    return binary_op(*args)
           ^^^^^^^^^^^^^^^^
jaxlib.xla_extension.XlaRuntimeError: RESOURCE_EXHAUSTED: Out of memory while trying to allocate 2319450112 bytes.
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
